modules/geo-layers/src/mvt-layer/mvt-layer.js

```diff
@@ -30,12 +34,68 @@ export default class MVTLayer extends TileLayer {
 
     const modelMatrix = new Matrix4().translate([xOffset, yOffset, 0]).scale([xScale, yScale, 1]);
 
+    props.autoHighlight = false;
     props.modelMatrix = modelMatrix;
     props.coordinateSystem = COORDINATE_SYSTEM.CARTESIAN;
     props.extensions = [...(props.extensions || []), new ClipExtension()];
 
     return super.renderSubLayers(props);
   }
+
+  onHover(info, pickingEvent) {
+    const {uniqueIdProperty, autoHighlight} = this.props;
+
+    if (autoHighlight) {
+      const {highlightedFeatureId} = this.state;
+      const hoveredFeature = info.object;
+      let hoveredFeatureId;
+
+      if (hoveredFeature) {
+        hoveredFeatureId = getFeatureUniqueId(hoveredFeature, uniqueIdProperty);
+      }
+
+      if (hoveredFeatureId !== highlightedFeatureId) {
+        this.setState({highlightedFeatureId: hoveredFeatureId});
+      }
+    }
+
+    return super.onHover(info, pickingEvent);
+  }
+
+  getHighlightedObjectIndex(tile) {
+    const {highlightedFeatureId} = this.state;
+    const {uniquePropertyId, highlightedObjectIndex: highlightedFeatureProp} = this.props;
+    const {data} = tile;
+
+    if ((!highlightedFeatureId && !highlightedFeatureProp) || !Array.isArray(data)) {
+      return -1;
+    }
+
+    const featureIdToHighlight = highlightedFeatureProp || highlightedFeatureId;
+
+    return data.findIndex(
+      feature =>
+        feature.id
+          ? feature.id === featureIdToHighlight
+          : feature.properties[uniquePropertyId] === featureIdToHighlight
+    );
+  }
+}
+
+function getFeatureUniqueId(feature, uniquePropertyId) {
+  const hasFeatureId = Boolean(feature.id);
+  const hasUniquePropertyId = Boolean(uniquePropertyId);
+
+  if (hasFeatureId) {
```

Comment:
I'd expect the user-specified `hasUniquePropertyId` to override the default

Reply:
Sure, makes more sense. I have just changed it.